test/backend_sum.in.cpp

```diff
@@ -403,6 +403,32 @@ NGRAPH_TEST(${BACKEND_NAME}, sum_3d_to_scalar)
               read_vector<float>(result));
 }
 
+NGRAPH_TEST(${BACKEND_NAME}, sum_3d_to_scalar_int32)
+{
+    Shape shape_a{3, 3, 3};
+    auto A = make_shared<op::Parameter>(element::i32, shape_a);
+    Shape shape_rt{};
+    auto f = make_shared<Function>(make_shared<op::Sum>(A, AxisSet{0, 1, 2}), ParameterVector{A});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
+
+    // Create some tensors for input/output
+    auto a = backend->create_tensor(element::i32, shape_a);
+    copy_data(
+        a,
+        vector<int32_t>{
+            0x40000001, 10, 19, 4,  13, 22, 7,  16, 25, 2,  11, 20, 5, 14,
+            23,         8,  17, 26, 3,  12, 21, 6,  15, 24, 9,  18, 27}); // 1073742202 == 0x4000017A
+    auto result = backend->create_tensor(element::i32, shape_rt);
+
+    auto handle = backend->compile(f);
+    handle->call_with_validate({result}, {a});
+    EXPECT_EQ((vector<int32_t>{0x40000001 + 10 + 19 + 4 + 13 + 22 + 7 + 16 + 25 + 2 + 11 + 20 + 5 +
+                               14 + 23 + 8 + 17 + 26 + 3 + 12 + 21 + 6 + 15 + 24 + 9 + 18 +
+                               27}), // NNP output 1073741824 == 0x40000000
```

Comment:
Is this comment about NNP left in nGraph intentionally? I would guess we should remove it.

Reply:
Yes ...this was intentional but I can remove it.